orm/src/test/java/org/hibernate/search/test/query/dsl/SortDSLTest.java

```diff
@@ -0,0 +1,251 @@
+		// set to true to display results
+		if ( outputLogs ) {
+			StringBuilder builder = new StringBuilder( "Initial coffee: " )
+					.append( originalInstance ) .append( "\n\n" )
+					.append( "Query: " ).append( mltQuery.toString() ).append( "\n\n" )
+					.append( "Matching coffees" ).append( "\n" );
+			for ( Object[] entry : results ) {
+				builder.append( "    Score: " ).append( entry[1] );
+				builder.append( " | Coffee: " ).append( entry[0] ).append( "\n" );
+			}
+			log.debug( builder.toString() );
+		}
+	}
+
+	private void indexTestData() {
+		Transaction tx = fullTextSession.beginTransaction();
+		final Calendar calendar = Calendar.getInstance( TimeZone.getTimeZone( "UTC" ), Locale.ROOT );
+		calendar.set( 1900, 2, 12, 0, 0, 0 );
+		calendar.set( Calendar.MILLISECOND, 0 );
+		Date january = calendar.getTime();
+		fullTextSession.persist(
+				new Month(
+						"January",
+						1,
+						"Month of colder and whitening",
+						"Historically colder than any other month in the northern hemisphere",
+						january,
+						0.231d,
+						"jan",
+						"<escaped>Month</escaped> of <em>colder</em> and <strong>whitening</strong>"
+				)
+		);
+		calendar.set( 100 + 1900, 2, 12, 0, 0, 0 );
+		february = calendar.getTime();
+		fullTextSession.persist(
+				new Month(
+						"February",
+						2,
+						"Month of snowboarding",
+						"Historically, the month where we make babies while watching the whitening landscape",
+						february,
+						0.435d,
+						"feb",
+						"Month of <em>snowboarding</em>"
+				)
+		);
+		calendar.set( 1800, 2, 12, 0, 0, 0 );
+		Date march = calendar.getTime();
+		fullTextSession.persist(
+				new Month(
+						"March",
+						3,
+						"Month of fake spring",
+						"Historically, the month in which we actually find time to go snowboarding.",
+						march,
+						0.435d,
+						"-mar",
+						"Month of <strong>fake</strong> spring"
+				)
+		);
+
+		Car car = new SportsCar( 1, "Leyland", 100 );
+		fullTextSession.persist( car );
+
+		car = new SportsCar( 2, "Morris", 180 );
+		fullTextSession.persist( car );
+
+		Day day = new Day( 1, 1 );
+		fullTextSession.persist( day );
+
+		day = new Day( 2, 2 );
+		fullTextSession.persist( day );
+
+		CoffeeBrand brand = new CoffeeBrand();
+		brand.setName( "Tasty, Inc." );
+		fullTextSession.persist( brand );
+
+		Coffee coffee = new Coffee();
+		coffee.setName( "Peruvian Gold" );
+		coffee.setBrand( brand );
+		fullTextSession.persist( coffee );
+
+		tx.commit();
+		fullTextSession.clear();
+	}
+
+	@Override
+	public Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] {
+				Month.class,
+				Car.class,
+				SportsCar.class,
+				Animal.class,
+				Day.class,
+				CoffeeBrand.class,
+				Coffee.class
+		};
+	}
+
+	@Override
+	public void configure(Map<String,Object> cfg) {
+		cfg.put( Environment.MODEL_MAPPING, MappingFactory.class.getName() );
```

Comment:
unrelated, but FYI you can put the `SearchMapping` instance directly in the cfg, so to not have to fiddle with classname, @Factory annotations, etc..